backend/inspirehep/accounts/api.py

```diff
@@ -66,3 +75,32 @@ def get_current_user_orcid():
 
 def get_current_user_remote_orcid_account():
     return RemoteAccount.query.filter_by(user_id=current_user.get_id()).one_or_none()
+
+
+def check_permissions_for_private_collections(
+    roles=[], collections=None, readwrite=False
```

Comment:
* :warning: [mutable default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) for `roles`, this is almost certainly not what you want. 
* Also it's not clear how this function is supposed to be used, what is the supplied `roles` supposed to be? Does it actually need to be parametric or is it always set to `[Roles.cataloger.value]`? 
* I also have the impression that this function is doing too many things and is mixing up the allowed roles for the collections and the user roles, I would split it into logical parts, which would be something like
    + define `get_allowed_roles_for_collections(collections, read_only=True)`: returns the union of all allowed roles for all collections, `read_only=True` returns `-read` and `read-write`, `False` returns only `-read-write` (this includes `superuser` and `cataloger`.
    + use that to intersect with the current user roles and give a result
    + then define two non-parametric `can_user_read_collections`/`can_user_edit_collections` that take only a `collections` arg.

Reply:
Yes, splitting the function is a good idea. When `read_only=True` in `get_allowed_roles_for_collections`, if it returns both `-read` and `-read-write`, users with `-read` role can have the write permission. Is it what we want? Or do you suggest to check all roles in the intersection?

Reply:
we allow if there's a non-empty intersection between the user roles and the allowed roles. The `read-write` role allows to read, so `read_only=True` (corresponding to a `read` operation) should also include `read-write`, in order for a user with `read-write` to be allowed to read. Or am I misunderstanding something?